### ISSUE TITLE
Picklist problems when there are spaces in the values

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#4536]
**[Mingle Card: 4876](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4876)**
Analysis
--------

Multi word picklist values for segment on eform config and Job worked fine and matched up. Couldn’t reproduce the issue listed below.

Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**| |
|**Hardware**| |
|**OS**| |
|**Browser**| |
|**Username**| |
|**Password**|Use LastPass|
|ORG ID| |
|User ID| |
|RayGun Error ID| |
  
 

```

Reported by Mike Mullins

I am trying to document something in Mingle but I can’t really walk you through the steps to duplicate without undoing a bunch of configuration due to the issue arising from the API name of a pick list field value.  I am hoping you can give me some guidance on how you would like me to document this issue.

So, the main issue is around the Segment field on the eForm Config object and the way it handles pick list values that are more than one word. 

· When you create a multi-word value for the Segment pick list on Job, it creates the API name with spaces instead of underscores.

· When you create the same pick list value in the picklist on the eForm object you get the same result.

· Leaving it this way results in no Forms tab at all in mobile under Job.

· You have to edit the API name for the multi-word pick list value on both objects to include the underscores for it to work in mobile.

· We have this same issue with Status Workflow on the SObject record type field when you use a multi-word value.

· Tickets have no concept of Segment by default, so I will need to add a formula field to look at Job Segment field and pull that value down so I can define it in the Ticket eForm configs.  I am assuming this issue won’t arise under that scenario but I’ll have to configure it and see what I get out of it.

Since I have already added the underscores to the API names for those values in the Segment fields on Job and eForm Config, I can’t undo that easily so as to list those steps in Mingle.  This is because you cannot Save a record if the API name does not contain underscores, so I would have to delete those pick list values from both objects and then have you re-add them to see this happen, which I can do if needed.  Do you need me to go through all of that to document the steps  to produce in Mingle or does this give you enough info to work off of?

This is for GR Energy specifically, but the issue will affect all orgs.  Please let me know what other info I can provide to assist.

MR:  Synced up with Mike. if both segment fields across job and eform config match via it’s api name, then the forms tab shows up fine. We use the api name for the picklist value rather then the actual value, which is fine as we have a history with other fields of using DeveloperName rather then Name, or Name rather then label. We did observe a salesforce bug that added some confusion to this card. If you change the api name on one of the segment fields from a multi word to a single word using underscores and save, then attempt to modify the api name to include spaces at a late date/time, salesforce throws an error and prevents the same.

To summarize, salesforce allows you to create a picklist value with a multi word api name. However, if you change the api name to a single word and save. It doesn’t then allow you go from a single word api name back to a multi word api name.  There’s a clear workaround for this(just make sure both segment field api value matches on both eform config and job segment fields), and a fix would have to be implemented by salesforce.

Expected Result
---------------

_What should happen?_

Actual Result
-------------

_What actually happens?_

_Analysis_
----------

Related Cards
-------------

Test Plan
---------
